packages/core/src/hooks/useHealthFactor.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react'
+import { useQuery } from '@apollo/client'
+
+import { bigNumberToNumber } from '../utilities'
+import { HealthFactorQuery } from '../graphql'
+
+import {
+  GET_HEALTH_FACTOR_DATA,
+  HealthFactorGraph,
+  UserReserveVariables,
+} from '@bloxifi/core'
+
+/**
+ * Options that can be used to configure useHealthFactor() hook.
+ */
+export interface Props {
+  /**
+   * User wallet account
+   */
+  readonly currentAccount?: string
+}
+
+/**
+ * Hook that returns calculated Health Factor value
+ */
+export const useHealthFactor = ({ currentAccount }: Props = {}): number => {
+  const [value, setValue] = useState(undefined)
+
+  const { data } = useQuery<HealthFactorGraph, UserReserveVariables>(
+    GET_HEALTH_FACTOR_DATA,
+    {
+      variables: {
+        user: currentAccount?.toLowerCase(),
+      },
+      onCompleted: () => calculate(),
+    },
+  )
+
+  const calculateTotalBorrow = (array: HealthFactorQuery[]): number =>
+    array.reduce(function (acc, { currentTotalDebt, reserve: { price } }) {
+      return (
+        acc +
+        bigNumberToNumber(currentTotalDebt) *
+          bigNumberToNumber(price.priceInEth)
+      )
+    }, 0)
+
+  const calculateTotalCollateralWithLT = (array: HealthFactorQuery[]): number =>
+    array.reduce(function (
+      acc,
+      { currentATokenBalance, reserve: { price, reserveLiquidationThreshold } },
+    ) {
+      return (
+        acc +
+        bigNumberToNumber(currentATokenBalance) *
+          bigNumberToNumber(price.priceInEth) *
+          reserveLiquidationThreshold *
+          Math.pow(10, -4)
+      )
+    },
+    0)
+
+  const calculate = () => {
+    const totalBorrowETH = calculateTotalBorrow(data.userReserves)
+    const totalCollateralETH = calculateTotalCollateralWithLT(data.userReserves)
+
+    setValue(totalCollateralETH / totalBorrowETH)
+  }
+
```

Comment:
we should add 
```tsx
useEffect(()=>{
useQuery<HealthFactorGraph, UserReserveVariables>(
    GET_HEALTH_FACTOR_DATA,
    {
      variables: {
        user: currentAccount?.toLowerCase(),
      },
      onCompleted: (data) => {
    const totalBorrowETH = calculateTotalBorrow(data.userReserves)
    const totalCollateralETH = calculateTotalCollateralWithLT(data.userReserves)

    setValue(totalCollateralETH / totalBorrowETH)
  },
    },
  )
},[currentAccount])
```

Reply:
this might not work since `useQuery` is also a hook. So React might complain, if so, then just remove the `useEffect` and `if(currentAccount) {`

Reply:
Yes, I got an error because of the useEffect, but why do we need it here? 
Health factor is updated if user switches the account